rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPool.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ * <p>
+ * <p>
+ * NOTICE: THIS FILE HAS BEEN MODIFIED BY  Qubole Inc UNDER COMPLIANCE WITH THE APACHE 2.0 LICENCE FROM THE ORIGINAL WORK
+ * OF https://github.com/DanielYWoo/fast-object-pool.
+ */
+package com.qubole.rubix.spi.fop;
+
+import java.net.SocketException;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Daniel
+ */
+public class ObjectPool<T>
+{
+  private final PoolConfig config;
+  private final ObjectFactory<T> factory;
+  private final ObjectPoolPartition<T>[] partitions;
+  private Scavenger scavenger;
+  private volatile boolean shuttingDown;
+
+  public ObjectPool(PoolConfig poolConfig, ObjectFactory<T> objectFactory)
+  {
+    this.config = poolConfig;
+    this.factory = objectFactory;
+    this.partitions = new ObjectPoolPartition[config.getPartitionSize()];
+    if (config.getScavengeIntervalMilliseconds() > 0) {
+      this.scavenger = new Scavenger();
+      this.scavenger.start();
+    }
+  }
+
+  public void registerHost(String host, int socketTimeout, int connectTimeout, int index)
+  {
+    partitions[index] = new ObjectPoolPartition<>(this, index, config, factory, createBlockingQueue(config), host, socketTimeout, connectTimeout);
+  }
+
+  protected BlockingQueue<Poolable<T>> createBlockingQueue(PoolConfig poolConfig)
+  {
+    return new ArrayBlockingQueue<>(poolConfig.getMaxSize());
+  }
+
+  public Poolable<T> borrowObject(int partitionNumber)
+          throws SocketException
+  {
+    return borrowObject(true, partitionNumber);
+  }
+
+  public Poolable<T> borrowObject(boolean blocking, int partitionNumber)
+          throws SocketException
+  {
+    Log.info("Borrowing object for partition: ", partitionNumber);
```

Comment:
change log level to debug for all pooling classes unless error/warnings are logged